src/app/component/CartItem/CartItem.component.js

```diff
@@ -284,10 +291,14 @@ export class CartItem extends PureComponent {
               mods={ { isLikeTable } }
             >
                 { this.renderOutOfStockMessage() }
-                { this.renderProductName() }
+                <div block="CartItem" elem="HeadingWrapper">
+                    { this.renderProductName() }
+                    { isMobile.any() ? this.renderDeleteButton() : null }
```

Comment:
Instead of ternary operator use `renderThing` pattern and check there whether this component part should be rendered or not.